### PR TITLE
overwrite multislogger.Logger memory rather than reassigning pointer

### PR DESCRIPTION
**Pull request by James-Pickett**
Noticed that some logs in main func were not getting shipped. When adding a handler we can overwrite the the memory address of the multslogger.Logger to make handlers propagate to previously assigned Loggers.

### CODE REVIEW

File: ee/agent/knapsack/knapsack.go [hunk 1]
```diff
@@ -51,7 +51,7 @@ func New(stores map[storage.Store]types.KVStore, flags types.Flags, db *bbolt.DB
 	}
 
 	if k.slogger != nil {
-		k.slogger.Logger = k.slogger.Logger.With("launcher_run_id", k.launcherRunId)
+		*k.slogger.Logger = *k.slogger.Logger.With("launcher_run_id", k.launcherRunId)
```
Comment (directionless) [inline ee/agent/knapsack/knapsack.go:54]: This spot feels like it breaks an abstraction. Somehow it feels wrong for knapsack to be reaching into slogger.Logger. But 🤷 not the end of the world

**Review by directionless — DISMISSED** [review]
seems okay to try

File: ee/agent/knapsack/knapsack.go (hunk 1 above)
Comment (James-Pickett) [reply ee/agent/knapsack/knapsack.go:54]: I agree, it's a weird spot. Knapsack needs access to the multislogger to be able to add handlers. We also want the launcher_run_id to be universally included in all logs ..... hmm maybe multislogger should just add it, then knapsack can stop reaching in there

Comment (James-Pickett) [reply ee/agent/knapsack/knapsack.go]: I moved this down to the multislogger, now knapsack is staying in it's lane and not reaching down into multislogger's guts

**Review by directionless — APPROVED** [review]
LGTM

**Review by RebeccaMahany — APPROVED** [review]

**Pull request merged**